src/jwtMiddleware.ts

```diff
@@ -88,53 +89,57 @@ export const jwtMiddleware = <
 >({
   key,
   algorithm,
-  critHandlers,
   customMessages = {},
   ignorePatterns,
-  onSuccess,
-  onFailure,
+  onSuccess = () => {},
+  onFailure = () => true,
 }: JwtMiddlewareOptions): T => {
   Object.assign(customMessages, errorMessages);
 
   const core: RouterMiddleware = async (ctx, next) => {
-    if (!ignorePatterns || !ignorePath(ctx, ignorePatterns)) {
-      let isUnauthorized = true;
-      let validatedJwt;
-      if (ctx.request.headers.has("Authorization")) {
-        const authHeader = ctx.request.headers.get("Authorization")!;
-
-        if (authHeader.startsWith("Bearer ") && authHeader.length > 7) {
-          const jwt = authHeader.slice(7);
-
-          validatedJwt = await validateJwt({
-            jwt,
-            key,
-            algorithm,
-            critHandlers,
-          });
-
-          if (validatedJwt.isValid) {
-            isUnauthorized = false;
-            onSuccess && onSuccess(ctx, validatedJwt);
-          }
-        }
+    const onUnauthorized = async (
+      jwtValidation: Error,
+    ) => {
+      const shouldThrow = onFailure(ctx, jwtValidation);
+      if (shouldThrow) {
+        ctx.throw(
+          Status.Unauthorized,
+          customMessages?.ERROR_INVALID_AUTH,
+        );
       }
 
-      if (isUnauthorized) {
-        if (onFailure) {
-          const ignoreFailure = onFailure(ctx, validatedJwt);
-
-          if (!ignoreFailure) return;
-        } else {
-          ctx.throw(
-            Status.Unauthorized,
-            customMessages?.ERROR_INVALID_AUTH,
-          );
-        }
-      }
+      await next();
+    };
+
+    // If request matches ignore, call next early
+    if (ignorePatterns && ignorePath(ctx, ignorePatterns)) {
+      await next();
+
+      return;
     }
 
-    await next();
+    // No Authorization header
+    if (!ctx.request.headers.has("Authorization")) {
+      await onUnauthorized(new Error("Authorization header is not present"));
```

Comment:
Let's use the error messages object here. Can you add "Authorization header is not present" to the object so that we can expose customization for the user?